ls1intum/Artemis
# Bugfix/programming exercise/c docker image as user

### Checklist

### Motivation and Context


Changes the C docker container in a way, it runs everything as user, not as `root`.
This fixes the issue related to bamboo not being able to delete docker files after usage.

### Description

* Running inside the C docker image not as `root`, but instead as user (`artemis_user`).
* Changed the  C template exercise from rotx to hello world as a more simplistic example for the tester.
* Updated the GBS-Tester from `1.27` to `1.31`.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new C programming exercise
4. Visit Bamboo
5. See, that Bamboo is now able to delete all files related to the docker run

### Test Coverage




### Screenshots
